app/uk/gov/hmrc/contactadvisors/controllers/SecureMessageController.scala

```diff
@@ -151,5 +213,30 @@ class CustomerAdviceAudit @Inject()(auditConnector: AuditConnector) {
         }
     }
   }
-}
 
+  def auditAdviceV2(result: Future[StorageResult])(implicit hc: HeaderCarrier): Unit = {
+    def createEvent(messageInfo: Map[String, String], auditType: String, transactionName: String) =
+      DataEvent(
+        auditSource = auditSource,
+        auditType = auditType,
+        tags = Map(EventKeys.TransactionName -> transactionName),
+        detail = Map(
+        ) ++ messageInfo
+      )
+
+    result.onComplete { res1 =>
+      res1.map {
+        case AdviceStored(messageId) => createEvent(Map("secureMessageId" -> messageId, "messageId" -> messageId), EventTypes.Succeeded, "Message Stored")
+        case _ => createEvent(Map("reason" -> s"Unexpected Error"), EventTypes.Failed, "Message Not Stored")
+      }.
+        recover { case ex =>
+          createEvent(Map("reason" -> s"Unexpected Error: ${ex.getMessage}"), EventTypes.Failed, "Message Not Stored")
+        }.
+        foreach { ev =>
+          auditConnector.sendEvent(ev).onFailure {
+            case err => Logger.error("Could not audit event", err)
```

Comment:
- I dont know what in 'err' but should not have personal data.
- error messages will generate pagerduty events, is this what we want ?

Reply:
This is a replication of the existing API behaviour. The new ticket will address new API auditing.